Update: made changes highlighted in the Google doc (#4)
codedex-io/projects: projects/visualize-youtube-data-with-plotly/visualize-youtube-data-with-plotly.mdx

display(df)
```

Let’s break down and understand what the `df = pd.read_csv(io.BytesIO(uploaded["most_subscribed_youtube_channels.csv"]))` piece of code is doing.

> We know that `uploaded` is a dictionary with file names as keys and file data as values. So, `uploaded["most_subscribed_youtube_channels.csv"]` is simply getting the binary data inside the file named `most_subscribed_youtube_channels.csv` that we uploaded to Google Colab.

Reviewer: ```suggestion
We know that `uploaded` is a dictionary with file names as keys and file data as values. So, `uploaded["most_subscribed_youtube_channels.csv"]` is simply getting the binary data inside the file named **most_subscribed_youtube_channels.csv** that we uploaded to Google Colab.
```

Python's `io` module manages file-related input and output operations. So, `io.BytesIO()` receives all the binary data and creates a stream of data. `pd.read_csv()` can't directly process binary data so we need to convert it into a stream before we can make a DataFrame from this data.

> This stream of data is finally read by `pd.read_csv()` method and turned into a DataFrame. We store this DataFrame in the `df` variable.

Reviewer: ```suggestion
This stream of data is finally read by the `pd.read_csv()` method and turned into a DataFrame. We store this DataFrame in the `df` variable.
```

Also, a quick shoutout to the channel all the way at the end that made it to 200 million ([T-Series](https://www.youtube.com/aashiqui2)). We already know their fan base is grinding.

> Plotly Express allows us to create much more customizable histograms by inputting a variety of arguments. Here's a link to [histogram documentation](https://plotly.com/python/histograms/) for further reference. In this section, we only used the base version with numerical data. But other histograms can also be created based on other columns in the YouTube data.

Reviewer: ```suggestion
Plotly Express allows us to create much more customizable histograms by inputting a variety of arguments. Here's a link to [histogram documentation](https://plotly.com/python/histograms/) for further reference. In this section, we only used the base version with numerical data. But other histograms can also be created using other columns in the YouTube data.
```
